server/models/users_test.go

```diff
@@ -14,43 +14,220 @@ import (
+// ユーザ一覧取得テスト
+func TestGetUsers(t *testing.T) {
+	//DB接続
+	db, err := connectDb()
 	if err != nil {
 		t.Errorf("failed to open mysql connection: %v", err)
 	}
-	ctx := context.Background()
+	//テーブル初期化
 	err = initUserTable(db)
 	if err != nil {
 		t.Errorf("failed to initUserTable: %v", err)
 	}
+	//テストユーザー作成
 	_, err = createUserForTest(db)
 	if err != nil {
 		t.Errorf("failed to initUserTable: %v", err)
 	}
+	//セッション登録
+	session := &pb.Session{
+		Uuid:   "test123",
+		Name:   Name,
+		Userid: 1,
+	}
+	err = CreateSession(session, db)
+	if err != nil {
+		t.Errorf("failed to CreateSession: %v", err)
+	}
+
+	//メタデータ設定
+	ctx := context.Background()
+	// m := map[string]string{"login_token": "test123"}
+	// md := metadata.New(m)
+	// ctx := metadata.NewIncomingContext(context.Background(), md)
 	request := pb.GetUsersRequest{}
 	users, err := GetUsers(ctx, db, request)
 	if err != nil {
 		t.Errorf("failed to GetUsers: %v", err)
 	}
+	err = bcrypt.CompareHashAndPassword([]byte(users[0].Password), []byte("abcd12341231"))
+	if err != nil {
+		t.Errorf("failed to Password: %v", err)
+	}
 	assert.Equal(t, users[0].Name, "テスト")
-	assert.Equal(t, users[0].Score, int32(123))
+	assert.Equal(t, users[0].Score, int32(1234))
 	assert.Equal(t, users[0].Photourl, "https://test")
 }
 
+//ユーザー登録テスト
+func TestCreateUser(t *testing.T) {
+	db, err := connectDb()
+	if err != nil {
+		t.Errorf("failed to open mysql connection: %v", err)
+	}
+	ctx := context.Background()
+	err = initUserTable(db)
+	if err != nil {
+		t.Errorf("failed to initUserTable: %v", err)
+	}
+	request := pb.CreateUserRequest{Name: Name, Score: Score, Photourl: Photourl, Password: Password}
+	_, err = CreateUser(ctx, db, request)
+	if err != nil {
+		t.Errorf("failed to CreateUser: %v", err)
+	}
+	getUsersRequest := pb.GetUsersRequest{}
+	users, err := GetUsers(ctx, db, getUsersRequest)
+	if err != nil {
+		t.Errorf("failed to GetUsers: %v", err)
+	}
+	err = bcrypt.CompareHashAndPassword([]byte(users[0].Password), []byte("abcd12341231"))
+	if err != nil {
+		t.Errorf("failed to Password: %v", err)
+	}
+	assert.Equal(t, users[0].Name, "テスト")
+	assert.Equal(t, users[0].Score, int32(1234))
+	assert.Equal(t, users[0].Photourl, "https://test")
+}
+
+//ユーザー取得テスト
+func TestGetUserById(t *testing.T) {
+	db, err := connectDb()
+	if err != nil {
+		t.Errorf("failed to open mysql connection: %v", err)
+	}
+	ctx := context.Background()
+	err = initUserTable(db)
+	if err != nil {
+		t.Errorf("failed to initUserTable: %v", err)
+	}
+	_, err = createUserForTest(db)
+	if err != nil {
+		t.Errorf("failed to initUserTable: %v", err)
+	}
+	getUsersRequest := pb.GetUsersRequest{}
+	users, err := GetUsers(ctx, db, getUsersRequest)
+	if err != nil {
+		t.Errorf("failed to GetUsers: %v", err)
+	}
+	fetchedUser, err := GetUserById(ctx, db, users[0].Id)
+	err = bcrypt.CompareHashAndPassword([]byte(fetchedUser[0].Password), []byte("abcd12341231"))
+	if err != nil {
+		t.Errorf("failed to Password: %v", err)
```

Comment:
一応、こう言ったエラーチェックの部分でも assert ライブラリ使えるので、試してみてください。😎
ちなみに、メッセージを一つ一つ考えるの面倒だと思うので、エラーチェックさえしていればなくても良いですよ。

```suggestion
		assert.NoError(err)
```

もしくは、厳密にいうと require パッケージを使うのが正しいので
（assert パッケージは失敗しても処理を継続しますが、require パッケージは panic で止まります）

```suggestion		
		require.NoError(err)
```

Reply:
ありがとうございます！
かしこまりました！